docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,298 @@
+from dataclasses import dataclass, field
+import uuid
+import itertools
+from typing import (
+    Generator,
+    Iterator,
+    Dict,
+    Sequence,
+    Optional,
+    TYPE_CHECKING,
+    Union,
+    Tuple,
+    List,
+)
+
+import uuid
+import scipy.sparse
+import weaviate
+
+from .... import Document
+from ..base.backend import BaseBackendMixin
+
+if TYPE_CHECKING:
+    from ....types import (
+        DocumentArraySourceType,
+    )
+
+
+@dataclass
+class WeaviateConfig:
+    """This class stores the config variables to initialize
+    connection to the Weaviate server"""
+
+    client: Optional[Union[str, weaviate.Client]] = None
+    n_dim: Optional[int] = None
+    name: Optional[str] = None
```

Comment:
In SqliteConfig, there's a parameter `serialize_config` and currently the serialization protocol is being changed to protobuf.
Let's add the same parameter here or create a `BaseConfig` class that has it and make both Config classes inherit from it

Reply:
Sure, `serialize_config` is made a parameter for now because different storages may have different serialization protocol by default (I'm using `pickle` in this case because currently `protobuf` is having issues with `to_base64`). While it's possible to also address this with `BaseConfig`, there may also be considerations with ordering kw/non-kw arguments and stuff. To make it simple and not to over-engineer, I just made it a parameter.